packages/api-fetch/src/index.js

```diff
@@ -36,6 +36,10 @@ function apiFetch( options ) {
 			headers[ 'Content-Type' ] = 'application/json';
 		}
 
+		if ( ! headers.Accept ) {
+			headers.Accept = 'application/json';
```

Comment:
Are all api-fetch requests `application/json` requests? I'm not sure about that? One I can think of is the metabox save request. I wonder if there's others?

Why is it necessary for the server side localization switching to have an 'Accept' header?

Reply:
> Are all api-fetch requests application/json requests? I'm not sure about that? One I can think of is the metabox save request. I wonder if there's others?

I'm not sure. I proposed this, or doing it in the locale middleware.

> Why is it necessary for the server side localization switching to have an 'Accept' header?

The server side localization requires the Accept header because it uses `wp_is_json_request()`. Core does not currently have a method to check whether the current request is a REST request early enough for the locale switching to work and adding a reliable one would be quite difficult.

Reply:
> The server side localization requires the Accept header because it uses `wp_is_json_request()`.

Maybe a crazy question but can't we just do the switch for all requests :) and not only REST requests? Seems like a `_locale` argument is explicit enough for all requests :) 

I'd understand the hesitation though and would be ok with the change but we should just test it properly. It's also unfortunate because this means we can't have APIs returning something else that is not JSON?

Reply:
> Maybe a crazy question but can't we just do the switch for all requests :) and not only REST requests? Seems like a _locale argument is explicit enough for all requests :)

Something for @pento and @danielbachhuber. I image we wouldn't really want to touch front-end localization as that opens up a whole new can of worms, but ¯\\\_(ツ)\_/¯.

>  It's also unfortunate because this means we can't have APIs returning something else that is not JSON?

We could adjust the `Accept` header slightly to `Accept: application/json, */*;q=0.1` that would indicate we'd prefer a JSON response, but fallback to everything. `wp_is_json_request` just checks that `application/json` is present in the accept header.

Reply:
> > The server side localization requires the Accept header because it uses wp_is_json_request().
> 
> Maybe a crazy question but can't we just do the switch for all requests :) and not only REST requests? Seems like a _locale argument is explicit enough for all requests :)

This probably isn't a good idea.

Reply:
> Are all api-fetch requests `application/json` requests? I'm not sure about that? One I can think of is the metabox save request. I wonder if there's others?

The REST API is a JSON API, so I personally think it's perfectly reasonable to say "Yes, all requests are `application/json`" (or, more realistically, "should be").

Reply:
The question is whether `apiFetch` is limited to the REST API (which is not the case in our usage for the moment).